### PR TITLE
【iOS】Fix Alert userInterfaceStyle having no effect

### LINKED ISSUE #38852: [iOS]: Alert userInterfaceStyle having no effect
### Description

I should be able to customize the color scheme of `Alert` by providing a `userInterfaceStyle` option but it does not seem to work on iOS. This is using the `react-native-cli` to bootstrap the project. With expo [it seems to work](https://snack.expo.dev/n7uXEtGO5?platform=ios).

Problem:

https://github.com/facebook/react-native/assets/8538369/ca4620a9-8ea2-4342-8d4f-10de27f169a2

I tried setting "userInterfaceStyle": "dark" in `app.json` but no luck.

### React Native Version

0.72.3

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
  OS: macOS 13.4.1
  CPU: (8) arm64 Apple M1 Pro
  Memory: 92.56 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.16.0
    path: ~/.nvm/versions/node/v18.16.0/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 9.5.1
    path: ~/.nvm/versions/node/v18.16.0/bin/npm
  Watchman:
    version: 2023.07.10.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.9862592
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.20
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.3
    wanted: 0.72.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

### Steps to reproduce

```
import React from 'react';
import {View, StyleSheet, Button, Alert} from 'react-native';

const App = () => {
  const showAlertLight = () => {
    Alert.alert('Alert Title', 'My Alert Msg', [{text: 'OK'}], {
      userInterfaceStyle: 'light',
    });
  };

  const showAlertDark = () => {
    Alert.alert('Alert Title', 'My Alert Msg', [{text: 'OK'}], {
      userInterfaceStyle: 'dark',
    });
  };

  return (
    <View style={styles.container}>
      <Button title="Light Alert" onPress={showAlertLight} />
      <Button title="Dark Alert" onPress={showAlertDark} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
});

export default App;
```

### Snack, screenshot, or link to a repository

Working example (expo): https://snack.expo.dev/n7uXEtGO5?platform=ios
Not working using `react-native-cli`

### PR DESCRIPTION
**Pull request by zhongwuzw**
## Summary:

Fixes #38852

## Changelog:

[IOS] [FIXED] - Fix Alert userInterfaceStyle having no effect

## Test Plan:

Run the demo from #38852 .

### CODE REVIEW

**Comment by zhongwuzw** [comment]
@dmytrorykun I updated the commit to fix the lint.

File: packages/react-native/React/CoreModules/RCTAlertController.mm [hunk 1]
```diff
@@ -43,9 +43,13 @@ - (UIWindow *)alertWindow
...
 {
-  UIUserInterfaceStyle style = RCTSharedApplication().delegate.window.overrideUserInterfaceStyle
-      ? RCTSharedApplication().delegate.window.overrideUserInterfaceStyle
-      : UIUserInterfaceStyleUnspecified;
+  UIUserInterfaceStyle style = self.overrideUserInterfaceStyle;
+  if (style == UIUserInterfaceStyleUnspecified) {
+    UIUserInterfaceStyle style = RCTSharedApplication().delegate.window.overrideUserInterfaceStyle
+        ? RCTSharedApplication().delegate.window.overrideUserInterfaceStyle
+        : UIUserInterfaceStyleUnspecified;
+  }
```
Comment (cipolleschi) [inline packages/react-native/React/CoreModules/RCTAlertController.mm:51]: Internally, our linter is much more stricter than externally.
You are redeclaring the exact same `UIUserInterfaceStyle style` variable twice and our internal builds are failing. Could you please aplly the following suggestion? 🙏 
```suggestion
  UIUserInterfaceStyle style = self.overrideUserInterfaceStyle;
  if (style == UIUserInterfaceStyleUnspecified) {
    style = RCTSharedApplication().delegate.window.overrideUserInterfaceStyle
        ? RCTSharedApplication().delegate.window.overrideUserInterfaceStyle
        : UIUserInterfaceStyleUnspecified;
  }
```

**Review by cipolleschi — CHANGES_REQUESTED** [review]
I left a comment as we need to fix a double declaration that made our internal build fails.
Thank you for the time you are spending on this! 🙇

File: packages/react-native/React/CoreModules/RCTAlertController.mm (hunk 1 above)
Comment (zhongwuzw) [reply packages/react-native/React/CoreModules/RCTAlertController.mm]: @cipolleschi Done.

**Review by cipolleschi — APPROVED** [review]
Thank you so much for this!

**Review by cipolleschi — APPROVED** [review]
Thank you so much for this!

**Pull request closed without merging**